src/DynamoCoreWpf/ViewModels/Search/SearchViewModel.cs

```diff
@@ -924,15 +935,36 @@ public void SearchAndUpdateResults(string query)
             if (string.IsNullOrEmpty(query))
                 return;
 
-            //Passing the second parameter as true will search using Lucene.NET
-            var foundNodes = Search(query);
-            searchResults = new List<NodeSearchElementViewModel>(foundNodes);
+            if (enableSeachThreading)
+            {
+                // A new search should cancel any existing searches.
+                searchCancelTooken?.Cancel();
+                searchCancelTooken?.Dispose();
+                searchCancelTooken = new();
 
-            FilteredResults = searchResults;
+                // We run the searches on the thread pool to reduce the impact on the UI thread.
+                Task.Run(() =>
+                {
+                    return Search(query, searchCancelTooken.Token);
 
-            UpdateSearchCategories();
+                }, searchCancelTooken.Token).ContinueWith((t, o) =>
+                {
+                    // This continuation will execute on the UI thread (forced by using FromCurrentSynchronizationContext())
+                    searchResults = new List<NodeSearchElementViewModel>(t.Result);
 
-            RaisePropertyChanged("FilteredResults");
```

Comment:
does Filtered Results setter already raise this?

Reply:
the FilteredResults setter raises this prop change event